peps/pep-0718.rst

```diff
@@ -62,6 +64,17 @@ If function objects were subscriptable, however, a more specific type could be g
 
    reveal_type(factory[int](lambda x: "Hello World" * x))  # type is Foo[int]
 
+There are even cases where subclass relations make type inference impossible. However,
+if you can specialise the function type checkers can infer a meaningful type.
+
+.. code-block:: python
+
+   def foo(x: Sequence[T] | T) -> list[T]: ...
+
+   reveal_type(foo[bytes](b"hello"))
+
+Unsolvable type parameters
```

Comment:
```suggestion
Unsolvable type parameters
--------------------------
```
I think this should be a heading? Not sure if this is the right heading level though.